cli/src/main/java/org/infinispan/cli/commands/kubernetes/Create.java

```diff
@@ -77,29 +82,60 @@ public boolean isHelp() {
-         add(resource, "kind", "Infinispan");
-         add(resource, "metadata.name", name);
-         add(resource, "spec.replicas", replicas);
+         KubernetesClient client = KubernetesContext.getClient(invocation);
+         namespace = Kube.getNamespaceOrDefault(client, namespace);
+         GenericKubernetesResource infinispan = new GenericKubernetesResource();
+         infinispan.setKind(INFINISPAN_CLUSTER_CRD.getKind());
+         ObjectMeta metadata = new ObjectMeta();
+         metadata.setName(name);
+         metadata.setNamespace(namespace);
+         infinispan.setMetadata(metadata);
+         GenericKubernetesResource spec = new GenericKubernetesResource();
+         infinispan.setAdditionalProperty("spec", spec);
+         spec.setAdditionalProperty("replicas", replicas);
          if (exposeType != null) {
-            add(resource, "spec.expose.type", exposeType);
+            GenericKubernetesResource expose = new GenericKubernetesResource();
+            spec.setAdditionalProperty("expose", expose);
+            expose.setAdditionalProperty("type", exposeType);
+            switch (ExposeCompleter.Expose.valueOf(exposeType)) {
+               case LoadBalancer:
+                  if (exposePort == 0) {
+                     throw Messages.MSG.exposeTypeRequiresPort(exposeType);
+                  } else {
+                     expose.setAdditionalProperty("port", exposePort);
+                  }
+                  break;
+               case NodePort:
+                  if (exposePort == 0) {
+                     throw Messages.MSG.exposeTypeRequiresPort(exposeType);
+                  } else {
+                     expose.setAdditionalProperty("nodePort", exposePort);
+                  }
+                  break;
+            }
+            if (exposeHost != null) {
+               expose.setAdditionalProperty("host", exposeHost);
+            }
          }
-         if (exposeHost != null) {
-            add(resource, "spec.expose.host", exposeHost);
-         }
-         if (exposePort > 0) {
-            add(resource, "spec.expose.port", exposePort);
-         }
-
-         try {
-            client.customResource(INFINISPAN_CLUSTER_CRD).create(Kube.getNamespaceOrDefault(client, namespace), resource);
-            return CommandResult.SUCCESS;
-         } catch (IOException e) {
-            return CommandResult.SUCCESS;
+         if (encryptionType != null) {
+            GenericKubernetesResource security = new GenericKubernetesResource();
+            spec.setAdditionalProperty("security", security);
+            GenericKubernetesResource encryption = new GenericKubernetesResource();
+            security.setAdditionalProperty("endpointEncryption", encryption);
+            encryption.setAdditionalProperty("type", encryptionType);
+            switch (EncryptionCompleter.Encryption.valueOf(encryptionType)) {
+               case Secret:
+               case Service:
```

Comment:
In the case of Service it's not necessary for the secret to be  provided as this is added to the spec by the Operator.